tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -598,3 +598,120 @@ def validate_header_rpt_month_year(datafile, header, generate_error):
+                    num_errors += self.__validate_ssp_s1_case(num_errors)
+                elif self.program_type == "SSP" and self.section == "C":
+                    num_errors += self.__validate_ssp_s2_case(num_errors)
+                else:
+                    self.total_cases_validated -= 1
+                    logger.warn(f"Case: {self.current_case} has no errors but has either an incorrect program type: "
+                                f"{self.program_type} or an incorrect section: {self.section}. No validation occurred.")
+                    self.has_validated = False
+            else:
+                logger.debug(f"Case: {self.current_case} has errors associated with it's records. "
+                             "Skipping Cat4 validation")
+        return num_errors
+
+    def __validate_tanf_s1_case(self, num_errors):
+        """Perform TANF Section 1 category four validation on all cached records."""
+        num_errors += self.__validate_tanf_s1_header_with_records()
+        return num_errors
+
+    def __validate_tanf_s2_case(self, num_errors):
+        """Perform TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s1_case(self, num_errors):
+        """Perform Tribal TANF Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s2_case(self, num_errors):
+        """Perform Tribal TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s1_case(self, num_errors):
+        """Perform SSP Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s2_case(self, num_errors):
+        """Perform SSP Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tanf_s1_header_with_records(self):
+        """Header YEAR + header QUARTER must be consistent with RPT_MONTH_YEAR for all T1, T2, and T3 records."""
+        year = self.header["year"]
+        quarter = self.header["quarter"]
+        header_rpt_month_year_list = get_rpt_month_year_list(year, quarter)
+        num_errors = 0
+        for record in self.records:
+            if record.RPT_MONTH_YEAR not in header_rpt_month_year_list:
+                num_errors += 1
+                err_msg = (f"Fail for RecordType={record.RecordType} and CASE_NUMBER="
+                           f"{record.CASE_NUMBER}. If YEAR={year} and QUARTER={quarter}, "
+                           "then RPT_MONTH_YEAR must be in "
+                           f"{header_rpt_month_year_list}.")
+                self.generated_errors.append(
+                    self.generate_error(schema=None,
+                                        error_category=ParserErrorCategoryChoices.CASE_CONSISTENCY,
+                                        error_message=err_msg,
+                                        record=None,
```

Comment:
can/should the record be added to the ParserError here?

Reply:
I think so. Also I can capture the main field that is causing the error.